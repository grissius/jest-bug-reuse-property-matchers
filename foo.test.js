const timestampMatcher = { createdAt: expect.any(String) };

const fooMatcher = {
  foo: timestampMatcher,
  bar: timestampMatcher
};

const fooMatcherSpread = {
  foo: { ...timestampMatcher },
  bar: { ...timestampMatcher }
};

const value = {
  foo: { createdAt: "def" },
  bar: { createdAt: "def", baz: "ab1" }
};
test("Matches property matchers", () => {
  expect(value).toMatchSnapshot(fooMatcher);
});

test("Matches property matchers - spread", () => {
  expect(value).toMatchSnapshot(fooMatcherSpread);
});
