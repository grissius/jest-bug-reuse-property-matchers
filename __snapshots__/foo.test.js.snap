// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches property matchers - spread 1`] = `
Object {
  "bar": Object {
    "baz": "ab1",
    "createdAt": Any<String>,
  },
  "foo": Object {
    "createdAt": Any<String>,
  },
}
`;

exports[`Matches property matchers 1`] = `
Object {
  "bar": Object {
    "baz": "ab1",
    "createdAt": Any<String>,
  },
  "foo": Object {
    "createdAt": Any<String>,
  },
}
`;
